src/components/SettingsPrompt/SettingsPrompt.vue

```diff
@@ -0,0 +1,284 @@
+                                            "
+                                        ></v-icon>
+                                    </template>
+
+                                    Copy to clipboard
+                                </v-tooltip>
+                            </template>
+                            <template v-slot:prepend>
+                                <v-tooltip location="bottom">
+                                    <template v-slot:activator="{ props }">
+                                        <v-icon
+                                            v-bind="props"
+                                            icon="mdi-refresh"
+                                            @click="regenerateAccessToken"
+                                        ></v-icon>
+                                    </template>
+
+                                    Generate new token
+                                </v-tooltip>
+                            </template>
+                        </v-text-field>
+                    </v-list-item-title>
+                </v-list-item>
+            </v-list>
+
+            <v-divider />
+
+            <!-- Switches for data -->
+            <v-list lines="two">
+                <v-list-subheader>Share</v-list-subheader>
+                <v-list-item
+                    class="pb-0 pt-0"
+                    title="Captured Data"
+                    subtitle="The server will have access to the streamed data"
+                >
+                    <template v-slot:prepend>
+                        <v-switch
+                            v-model="streamData"
+                            class="mr-4"
+                            color="primary"
+                            inset
+                        ></v-switch>
+                    </template>
+                </v-list-item>
+
+                <v-list-item
+                    class="pb-0 pt-0"
+                    title="Regex & Capture Area Settings"
+                    subtitle="The server will have access to the regex and capture area settings"
+                >
+                    <template v-slot:prepend>
+                        <v-switch
+                            v-model="streamRegexAndCaptureAreaSettings"
+                            class="mr-4"
+                            color="primary"
+                            inset
+                        ></v-switch>
+                    </template>
+                </v-list-item>
+            </v-list>
+        </v-card>
+    </v-dialog>
+</template>
+
+<script setup lang="ts">
+import useNotificationSystem from '@/composables/useNotificationSystem/useNotificationSystem';
+import useClipboard from '@/composables/useClipboard/useClipboard';
+import useTokenGenerator from '@/composables/useTokenGenerator/useTokenGenerator';
+import { onMounted, ref } from 'vue';
+
+onMounted(() => {
+    // generate a new access token on application start and validate it
+    regenerateAccessToken();
```

Comment:
We should do this on opening of the session settings. It seems kind of misleading to get the "session token valid" notification right after opening the application